src/errors.js

```diff
@@ -1,11 +1,52 @@
-export class PreconditionFailed extends Error {
+export class ScnnrError extends Error {
   constructor(message) {
     super(message)
-    this.name = 'PreconditionFailed'
+    this[Symbol.toStringTag] = 'scnnr-error'
     if (Error.hasOwnProperty('captureStackTrace')) {
-      Error.captureStackTrace(this, PreconditionFailed)
+      Error.captureStackTrace(this, ScnnrError)
     } else {
       this.stack = (new Error()).stack
     }
   }
 }
+
+export class PreconditionFailed extends ScnnrError {
+  constructor(message) {
+    super(message)
+    this.name = 'PreconditionFailed'
+  }
+}
+
+export class ScnnrAPIError extends ScnnrError {
+  constructor({ title, message, statusCode, rawResponse }) {
+    super(message)
+    this[Symbol.toStringTag] = 'scnnr-api-error'
+    this.name = (title || 'ScnnrAPIError').replace(/ /g, '')
+    this.statusCode = statusCode
+    this.rawResponse = rawResponse
+  }
+}
+
+export class ForbiddenError extends ScnnrAPIError {
+  constructor({ rawResponse }) {
+    const message = 'You don\'t have access to this resource'
+    super({ title: 'Forbidden', message, rawResponse, statusCode: 403 })
+  }
+}
+
+export class TooManyRequestsError extends ScnnrAPIError {
+  constructor({ rawResponse }) {
+    const message = 'Exceeded request quota'
+    super({ title: 'TooManyRequests', message, rawResponse, statusCode: 429 })
+  }
+}
+
+const httpErrorsByCode = {
```

Comment:
That's true. I would think of this object more like a way to track "exceptions to the rule", in this we could list status codes that need special handling. 

Unless there are errors that use a certain status code but sometimes adhere to the error schema of the API, and sometimes not. 

For example:
404 response
```
{"detail":"Recognition (ID: some-id) is not found.","title":"Not Found","type":"not-found"}
```
other 404 response
```
{"message": "Not found"}
```

If this is something that could happen I rework a bit the solution to work properly with this case.

Reply:
We want to make all responses adhere the error schema of the API in future. But for now, there are some cases to return responses like as `{"message": "Not found"}`.

Actually, the behavior came from using AWS API Gateway. If a request reaches our API server, the response adheres the error schema.
On the other hand, in case that a request is refused by API Gateway, the response is generated by AWS. So it does not adhere its schema.
This can be determined by checking `Content-Type` in the response header.

Reply:
In addition, I think that the situations that a request is refused by API Gateway are abnormal cases for this library.
Because 404 responses from API Gateway are probably a bug of building URL by this library, and 403 responses are prevented by throwing `PreconditionFailed`.

So, we should mainly handle errors from our API server using custom error classes instead of errors from API Gateway. Because users will mostly face the errors from our API server.